syncer/syncer.go

```diff
@@ -684,6 +684,19 @@ func (s *Syncer) trackTableInfoFromDownstream(tctx *tcontext.Context, origSchema
 		createStmt.Table.Schema = model.NewCIStr(origSchema)
 		createStmt.Table.Name = model.NewCIStr(origTable)
 
+		// schema tracker sets non-clustered index, so can't handle auto_random.
```

Comment:
Although we have turned `tidb_enable_clustered_index` off in `defaultGlobalVars`, the user can turn it back on via the configuration file, is there a problem if we hard delete the `ColumnOptionAutoRandom` there?

Reply:
the schema tracker is mainly used to 

1. get column names because they are not written in binlog row events.
2. calculate if two row changing an be sent concurrently by PK/UK
3. calculate information of optimistic shard DDL

(don't know if I forget something, cc @GMHDBJD )

I think there's no harm to delete `ColumnOptionAutoRandom`, but agree we'd better reduce the modification. Will fix later